force.go

```diff
@@ -18,16 +18,17 @@ const (
 
 	logPrefix = "[simpleforce]"
 
-	duplicateRuleHeader = "Sforce-Duplicate-Rule-Header"
+	duplicateRuleHeader         = "Sforce-Duplicate-Rule-Header"
+	autoAssignRuleHeader string = "Sforce-Auto-Assign"
 )
 
 type Client interface {
 	Query(ctx context.Context, query, nextRecordsURL string) (*QueryResult, error)
 
 	DescribeSObject(ctx context.Context, sobj *SObject) (*SObjectMeta, error)
-	CreateSObject(ctx context.Context, sobj *SObject, blacklistedFields []string, allowDuplicates bool) error
+	CreateSObject(ctx context.Context, sobj *SObject, blacklistedFields []string, allowDuplicates bool, autoAssign *bool) error
 	GetSObject(ctx context.Context, sobj *SObject) error
-	UpdateSObject(ctx context.Context, sobj *SObject, blacklistedFields []string) error
+	UpdateSObject(ctx context.Context, sobj *SObject, blacklistedFields []string, autoAssign *bool) error
```

Comment:
This interface is a bit awkward now
I'm okay with it for now but I could see exposing these options as functional options

Reply:
I agree. Especially with the default behavior when omitting this new header.